gcloud-java-core/src/main/java/com/google/gcloud/BasePage.java

```diff
@@ -14,41 +14,49 @@
  * limitations under the License.
  */
 
-package com.google.gcloud.storage;
+package com.google.gcloud;
 
 import java.io.Serializable;
 import java.util.Collections;
-import java.util.Iterator;
 import java.util.Objects;
 
 /**
- * Base implementation for Google Cloud storage list result.
+ * Base implementation for Google Cloud paginated results.
  */
-public class BaseListResult<T extends Serializable> implements ListResult<T>, Serializable {
+public class BasePage<T> implements Page<T>, Serializable {
 
-  private static final long serialVersionUID = -6937287874908527950L;
+  private static final long serialVersionUID = 3914827379823557934L;
 
   private final String cursor;
   private final Iterable<T> results;
   private final NextPageFetcher<T> pageFetcher;
 
-  public interface NextPageFetcher<T extends Serializable> extends Serializable {
-    ListResult<T> nextPage();
+  public interface NextPageFetcher<T> extends Serializable {
+    Page<T> nextPage();
   }
 
-  public BaseListResult(NextPageFetcher<T> pageFetcher, String cursor, Iterable<T> results) {
+  /**
+   * Creates a {@code BasePage} object. In order for the object to be serializable the {@code
+   * result} parameter must be serializable.
```

Comment:
Why do you think it causes a javadoc error? I'll change it to `{@code results}` as I missed one `s`.